### DISCUSSION TITLE
Retrying workflows on non-transient errors and transient errors

### DISCUSSION THREAD

**Discussion by alotkros** [argoproj/argo-workflows#10716] [Q&A] [answered]
Hi,

I'm having trouble with workflows that have their retryPolicy set to `onTransientError`, that started to fail due to non-transient errors.
Supposedly, switching the retryPolicy to `onError` should work - however I can't manage to find information on if `onError` is a superset of `onTransientError` and covers all the transient errors as well?
I know that setting the retryPolicy to `Always` covers all errors - but I'm trying to specifically to not retry on failures.

**Comment by terrytangyuan** [comment, marked as answer]
Yes it's a superset. If you need to cover additional transient errors, you can set env var `TRANSIENT_ERROR_PATTERN`. https://github.com/argoproj/argo-workflows/blob/master/docs/environment-variables.md
